tekton/utils/suite_test.go

```diff
@@ -17,22 +17,59 @@ limitations under the License.
 package utils
 
 import (
+	"context"
 	"testing"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
 
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 	//+kubebuilder:scaffold:imports
 )
 
+var (
+	cfg       *rest.Config
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
 func Test(t *testing.T) {
 	RegisterFailHandler(Fail)
-
-	RunSpecs(t, "Tekton Utils Suite")
+	RunSpecs(t, "Tekton Suite")
 }
 
 var _ = BeforeSuite(func() {
 	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	// adding required CRDs, including tekton for PipelineRun Kind
+	testEnv = &envtest.Environment{}
+
+	var err error
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	//+kubebuilder:scaffold:scheme
+	k8sClient, err = client.New(cfg, client.Options{
```

Comment:
what do we need this for? It seems like the tests are just manipulating structs unless I missed something

Reply:
Removed.